UMN-VMWareRA.psm1

```diff
@@ -1197,6 +1197,321 @@ function Set-VMWareRAVMTagAttachedToObject {
+            {
+                $url += $urlFilterItem + "&";
+            }
+        }
+
+        $return = Invoke-WebRequest -Uri $url -Method Get -Headers @{'vmware-api-session-id'=$sessionID} -ContentType 'application/json' -UseBasicParsing
+        return(($return.Content | ConvertFrom-Json).value)
+    }
+    End
+    {
+    }
+}
+#endregion
+
+#region Get-VMWareRAVMByName
+function Get-VMWareRAVMByName {
+<#
+    .Synopsis
+        Get VM objects by name from the VMWare Rest API.  Supports wildcards and handles case sensitivity issues.
+
+    .DESCRIPTION
+        Get details about vm or a list of vms from VMWare Rest API.  Supports wildcards and handles case sensitivity issues.
+        Able to return more than 1000 objects by working around the current limitation of the REST APIs.
+
+    .PARAMETER vCenter
+        FQDN of server to connect to
+
+    .PARAMETER sessionID
+        vmware-api-session-id from Connect-vmwwarerasession
+
+    .PARAMETER name
+        name of the VM, case sensitivity not required. Supports wildcard character *.
+
+    .PARAMETER host
+        name of host to return list of VMs from, case sensitivity required
+
+    .PARAMETER hostID
+        ID of host to return list of VMs from. Overrides host parameter if also specified.
+
+    .PARAMETER cluster
+        Name of cluster to return list of VMs from (able to return more then 1000 VMs), case sensitivity required.
+
+    .PARAMETER clusterID
+        ID of cluster to return list of VMs from (able to return more then 1000 VMs.) Overrides cluster parameter if specified.
+
+    .OUTPUTS
+        List of VM objects based on parameters specified.
+
+    .Notes
+        Author: Aaron Smith
+#>
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter(Mandatory)]
+        [string]$vCenter,
+        
+        [Parameter(Mandatory)]
+        [string]$sessionID,
+
+        [string]$name,
+
+        [string]$host,
+
+        [string]$hostID,
+
+        [string]$cluster,
+
+        [string]$clusterID
+    )
+
+    Begin
+    {
+    }
+    Process
+    {
+        if ($cluster)
+        {
+            $clusterID = (Get-VMWareRACluster -vCenter $vCenter -sessionID $sessionID -name $cluster).cluster;
+
+            if (!$clusterID)
+            {
+                throw "Cluster ${cluster} not found by name";
+            }
+        }
+
+        if ($host)
+        {
+            $hostID = (Get-VMWareRAHost -vCenter $vCenter -sessionID $sessionID -name $host).host;
+
+            if (!$hostID)
+            {
+                throw "VMHost ${host} not found by name";
+            }
+        }
+
+        ## List of VMs compiled from queries for return
+        [Array] $vmList = @();
+
+        ## Construct url
+        $url = "https://$vCenter/rest/vcenter/vm"
+
+        [Array] $urlFilters = @();
+
+        if ($hostID)
+        {
+            $urlFilters += "filter.hosts=${hostID}";
+        }
+
+        if ($urlFilters.Count -gt 0)
```

Comment:
You've possibly got a clusterID, why isn't it added as a filter?

Reply:
Because filter.clusters will trigger the 1,000 VM limitation in many cases.  So instead, that's handled later in the code where cluster ID is used to get the host list.  Then the host list is used to get the list of VMs, one host at a time.

Reply:
Ok, sounds good.